beard/metrics/tests/test_text.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Test string metrics.
+
+.. codeauthor:: Petros Ioannidis <petros.ioannidis91@gmail.com>
+
+"""
+from __future__ import generators
+import pytest
+from pytest import mark
+from ..text import find_all, jaro_matching, jaro
+
+
+@mark.parametrize('s, letter, occur',
+                  (('MARTHA', 'A', (1, 5)),
+                   ('DWAYNE', 'D', (0, )),
+                   ('A', 'A', (0, )),
+                   ('AABAA', 'AA', (0, 3)),
+                   ('ABCD', 'D', (3, ))))
+def test_find_all_normal_string(s, letter, occur):
+    """Test find_all behaviour for average cases."""
+    assert tuple(find_all(s, letter)) == occur
+
+
+@mark.parametrize('s, letter',
+                  (('MARTHA', 'Z'),
+                   ('', 'A')))
+def test_find_all_none_string(s, letter):
+    """Test find_all behaviour for empty cases."""
+    with pytest.raises(StopIteration):
+        assert next(find_all(s, letter))
```

Comment:
Might be nice to use here `six` library:

http://docs.pythonsprints.com/python3_porting/py-porting.html#iterators
http://pythonhosted.org/six/#six.next